### DISCUSSION TITLE
No API endpoint to create reactions on PR review bodies

### DISCUSSION THREAD

**Discussion by samuelcolvin** [community/community#29018] [Apps, API and Webhooks]
There are endpoints to:
* ["Create reaction for an issue comment"](https://docs.github.com/en/rest/reactions#create-reaction-for-an-issue-comment) - this also works for normal comments on PRs
* ["Create reaction for a pull request review comment"](https://docs.github.com/en/rest/reactions#create-reaction-for-a-pull-request-review-comment)

But very weirdly there's no api endpoint to create reactions on pull request review bodies (e.g. the text you enter at the end of a review, before clicking submit), even though this is possible in the UI.

Why is this? Please could it be added.

**Comment by gr2m** [comment]
the same is true for retrieving a list of reactions. It's possible to get them for pull request review comments, but not for pull request reviews, at least not via the REST API

**Reply by xmo-odoo** [reply]
A complementary missing feature is that there is no `reactions` field on PR reviews, unlike both issue comments and pr review comments. That field contains tallies for each reaction type, and the URL to list and post reactions to the object.

(and that's more of an ancillary note, but there's also no good link between PR reviews and their review comments, you only have `pull_request_review_id`, there's no formatted link to follow, and there's nothing from review to review comment).

**Comment by sergiivolchkovatbetsson** [comment]
Still no API for this in 2025. 

The API for [pull request reviews](https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#list-comments-for-a-pull-request-review) describes Pull Request Reviews as 
> groups of pull request review comments on a pull request, grouped together with a state and optional body comment.

However, the "optional body comment" to a pull request review is not returned as comments to the PR, so the _issue comment_ API cannot be used. 
Pull request review comments API supports reactions, but those only apply to the diff-bound code comments as a part of the PR review.

The PR reviews themselves fall between the cracks.

My scenario: 

- Enable GitHub Copilot code reviews;
- Ask developers to react to the reviews with thumbs up or down to assess their usefulness - similarly to GitHub's own suggestion for developers to [react to review comments](https://docs.github.com/en/copilot/using-github-copilot/code-review/using-copilot-code-review#providing-feedback-on-copilots-reviews), but on the level of the review itself;
- Gather repo- and org-wide stats on the usefulness of these comments.

**Comment by cmelone** [comment]
It's unfortunate that this feature isn't available through the REST API, but it is possible via GraphQL: https://docs.github.com/en/graphql/reference/mutations#addreaction. Just add a reaction to the `node_id` of the review you're targeting.
